src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -52,35 +53,45 @@ trait Validation {
     override def apply(t: T): Result = if (!b(t)) Success else validator(t)
   }
 
-  implicit def every[T](implicit validator: Validator[T]): Validator[Iterable[T]] = {
-    new Validator[Iterable[T]] {
-      override def apply(seq: Iterable[T]): Result = {
-
-        val violations: Set[Violation] = seq match {
-          case m: Map[_, _] =>
-            m.map(item => (item, validator(item))).collect {
-              case ((k, v), f: Failure) => GroupViolation(k, "not valid", Some(s"($k)"), f.violations)
-            }(collection.breakOut)
-          case _ =>
-            seq.map(item => (item, validator(item))).zipWithIndex.collect {
-              case ((item, f: Failure), pos: Int) => GroupViolation(item, "not valid", Some(s"($pos)"), f.violations)
-            }(collection.breakOut)
+  implicit def everyKeyValue[T](implicit validator: Validator[T]): Validator[Iterable[(String, T)]] = {
+    def mapViolation(violation: Violation, desc: Description): Violation = {
```

Comment:
Is this where the description comes from now?

Reply:
I simplified this code; description is... a path component. Anyways next patch is clearer.